swizzin/swizzin
# test: Testing suite

## Description

Introduces testing to swizzin! 😲 

## Fixes issues: 
- N/A

## Proposed Changes:
- Add scripts for test
    - basic default test that will try all it can based off just an app name
    - reusable functions for tests
    - couple implementations and low hanging fruit
- Add `box test [test(s)]`
- Run `box test` in `setup.sh` when `--run-checks` is passed along
- Add testing check boxes to PR template

## Change Categories

- New feature 
- Change in `functions` 
- Project structure 
- Contribution flow 

## Checklist

- [ ] Docs have been made OR are not necessary
    - PR link: 
- [x] Changes to panel have been made OR are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Code conforms to project structure [(See more)](https://swizzin.ltd/dev/structure)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have functionally tested the changes being introduced

## Test scenarios


### Architectures

|   			| `amd64` 	| `armhf` 	| `arm64` 	| Unspecified 	|
|--------		|-------- 	|-------- 	|-------- 	|----------		|
| Focal 		|			|			|	✅	|				|
| Bionic		|			|			|			|				|
| Buster		|			|			|			|				|
| Stretch		|			|			|			|				|
| Raspbian  	|	⚫️		|			|	⚫️		|	⚫️			|

### ✅❎ Passed
<img width="1137" alt="image" src="https://user-images.githubusercontent.com/23618693/120080302-8272a900-c0b8-11eb-8b7d-7e8627ddddcc.png">
### 🛠🛠 TODO

### ❌❌ Currently failing
